storage-access.bs

```diff
@@ -105,23 +118,14 @@ A {{Document}} is in a <dfn>first-party-site context</dfn> if it is the [=active
 
 A {{Document}} is in a <dfn>third party context</dfn> if it is not in a [=first-party-site context=].
 
-<h3 id="ua-state">User Agent state related to storage access</h3>
-
-A <dfn>storage access map</dfn> is a [=map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
-
-User Agents maintain a single <dfn>global storage access map</dfn>.
-
-ISSUE(privacycg/storage-access#2): What is the lifecycle of the [=global storage access map=]? How long do we remember its contents? Firefox and Safari differ here.
+<h3 id="ua-state">Changes to user agent state related to storage access</h3>
 
-ISSUE(privacycg/storage-access#5): When do we age out entries in the [=global storage access map=]? See also [Scope of Storage Access](https://github.com/privacycg/storage-access#scope-of-storage-access).
+Modify the definition of [=environment=] in the following manner:
+1. Add a new member called <dfn for="environment">has storage access</dfn> of type [=boolean=].
 
-Each [=agent cluster=] has a <dfn for="agent cluster">storage access map</dfn>.
-
-When an [=agent cluster=] is created, its [=agent cluster/storage access map=] is initialized with a [=map/clone=] of the [=global storage access map=].
-
-To <dfn type="abstract-op">obtain the storage access map</dfn> for a {{Document}} |doc|, run the following steps:
-
-1. Return the [=agent cluster/storage access map=] of |doc|'s [=relevant agent=]'s [=agent cluster=].
+Modify the definition of [=source snapshot params=] in the following manner:
```

Comment:
Genuine rookie question: it is fine to modify the snapshot later?

Reply:
I might be misunderstanding your question, but this isn't mutating a snapshot; it's modifying the *definition* of snapshot such that every snapshot includes an additional two fields, the values of which are specified in the "Changes to navigation" section (under "When snapshotting source snapshot params"). Does that answer the question?

Reply:
Ah, nevermind, I realize the premise of my question was bad. I thought the updates to [=environment/has storage access=] in [=determine the storage access policy=] were referring to the snapshot, not the new bool on the environment. All good!